components/playground/main.go

```diff
@@ -147,19 +155,33 @@ func hasDashboard(pdAddr string) bool {
 	return false
 }
 
-func bootCluster(version string, pdNum, tidbNum, tikvNum int, host string, monitor bool) error {
+func bootCluster(version string, pdNum, tidbNum, tikvNum int, host string, monitor bool, tidbBinPath, tikvBinPath, pdBinPath string) error {
 	if pdNum < 1 || tidbNum < 1 || tikvNum < 1 {
 		return fmt.Errorf("all components count must be great than 0 (tidb=%v, tikv=%v, pd=%v)",
 			tidbNum, tikvNum, pdNum)
 	}
 
+	var pathMap = make(map[string]string)
```

Comment:
I think we should make further handling for the `*path` and convert it to an absolute path. 
According to https://github.com/pingcap-incubator/tiup/blob/e08d28e360c6aee5044a9308c74eb994281be4e4/cmd/run.go#L198. Them will be related to the playground work directory, which is not same as the working directory of the user running `tiup playground`.

We can achieve our intention via the following code snippet.
```
if !strings.HasPrefix(tidbBinPath, "/") && !strings.HasPrefix(pdConfigPath, "~") {
    tidbBinPath = filepath.Join(os.Getenv(localdata.EnvNameWorkDir), tidbBinPath)
}
```

Reply:
fixed, please look again.